features/step_definitions/collection.js

```diff
@@ -0,0 +1,275 @@
+            Assert.strictEqual(response.statusCode, 200);
+            Assert.strictEqual(response.body.name, 'myCollection');
+        });
+    });
+
+    Then(/^the collection contains that pipeline$/, function step() {
+        const pipelineId = parseInt(this.pipelineId, 10);
+
+        return request({
+            uri: `${this.instance}/${this.namespace}/collections/${this.firstCollectionId}`,
+            method: 'GET',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.strictEqual(response.statusCode, 200);
+            Assert.deepEqual(response.body.pipelineIds, [pipelineId]);
+        });
+    });
+
+    When(/^they create a new collection "myCollection"$/, function step() {
+        return createCollection.call(this, { name: 'myCollection' })
+            .then((response) => {
+                Assert.strictEqual(response.statusCode, 201);
+                this.firstCollectionId = response.body.id;
+            });
+    });
+
+    Then(/^the collection is empty$/, function step() {
+        return request({
+            uri: `${this.instance}/${this.namespace}/collections/${this.firstCollectionId}`,
+            method: 'GET',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.strictEqual(response.statusCode, 200);
+            Assert.deepEqual(response.body.pipelineIds, []);
+        });
+    });
+
+    When(/^they update the collection "myCollection" with that pipeline$/,
+        { timeout: TIMEOUT }, function step() {
+            return this.ensurePipelineExists({ repoName: this.repoName })
+                .then(() => {
+                    const pipelineId = parseInt(this.pipelineId, 10);
+
+                    request({
+                        uri: `${this.instance}/${this.namespace}/collections/` +
+                            `${this.firstCollectionId}`,
+                        method: 'PUT',
+                        auth: {
+                            bearer: this.jwt
+                        },
+                        body: {
+                            pipelineIds: [pipelineId]
+                        },
+                        json: true
+                    }).then((response) => {
+                        Assert.strictEqual(response.statusCode, 200);
+                    });
+                });
+        });
+
+    Given(/^they have a collection "myCollection"$/, function step() {
+        return createCollection.call(this, { name: 'myCollection' })
+            .then((response) => {
+                Assert.oneOf(response.statusCode, [409, 201]);
+
+                if (response.statusCode === 201) {
+                    this.firstCollectionId = response.body.id;
+                } else {
+                    const str = response.body.message;
+
+                    [, this.firstCollectionId] = str.split(': ');
```

Comment:
You won't be able to parse the id out from the error object.

Reply:
Good catch! I fixed it now on the API side of things to add the id in the message